parla/task_runtime.py

```diff
@@ -2218,7 +2221,7 @@ def launch_tasks_callback(self):
         # Check runtime conditions
         # Are there any tasks to launch?
         # Are there any free worker threads?
-        condition = len(self._free_worker_threads) > 0
+        condition = len(self._free_worker_threads) > 0 and self.num_active_tasks() != 0
```

Comment:
Everything looks good. 

This is the only part that confuses me though, if a worker thread has been assigned an active task it shouldn't be free. (e.g its popped off of the queue in _launch_task). Is this a redundant check?

Reply:
I think that problem has been resolved. A thread appends itself to the free list before starting the scheduler callback. Or is there anything that I am missing? Let me know then I will check it. For now, let me merge this PR.